src/components/Spotlight/Spotlight.tsx

```diff
@@ -275,32 +285,42 @@ const Spotlight = ({
   ]);
 
   // TODO: use a resize observer to detect when the bounds of the target change
-  const updateWindowBounds = () => {
+  const updateWindowBounds = useCallback(() => {
     setWindowDimensions({
       width: window.innerWidth,
       height: window.innerHeight,
     });
-  };
+  }, [setWindowDimensions]);
 
-  const updateScrollPosition = (e: Event) => {
+  const updateScrollPosition = useCallback(() => {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore documentElement does exist on target element
-    setScrollTop(e?.target?.documentElement.scrollTop);
-  };
+    // setScrollTop(e?.target?.documentElement.scrollTop);
+    setScrollTop(window.scrollY);
+  }, [setScrollTop]);
+
+  useEffect(() => {
+    updateScrollPosition(); // update scroll position every render that the scroll changes so the previous value will update after scrolling events stop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [window.scrollY]);
```

Comment:
I'm not confident that deps will react to changes on `window.xxxx` so I don't think this is doing what you expect.